src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/HashBuilder.cs

```diff
@@ -28,25 +45,29 @@ public static string BuildIDFromSymbol(ITypeSymbol typeSymbol)
 		/// <param name="input">The string to hash</param>
 		/// <param name="algoritm">The algorithm to use</param>
 		/// <returns>A hex-formatted string of the hash</returns>
-		public static string Build(string input, HashAlgorithm algoritm = null)
+		public static string Build(string input)
 		{
-			using (algoritm ??= SHA256.Create())
+			using (var algoritm = SHA256.Create())
 			{
 				// Convert the input string to a byte array and compute the hash.
 				var data = algoritm.ComputeHash(Encoding.UTF8.GetBytes(input));
 
-				// Create a new Stringbuilder to collect the bytes
-				// and create a string.
-				var builder = new StringBuilder();
-
-				// Use the first 16 bytes of the hash
-				for (int i = 0; i < Math.Min(data.Length, 16); i++)
+				var min = Math.Min(data.Length, 16);
+				var result = new string((char)0, min * 2);
+				unsafe
 				{
-					builder.Append(data[i].ToString("x2", CultureInfo.InvariantCulture));
+					fixed (char* resultP = result)
+					{
+						for (int i = 0; i < min; i++)
+						{
+							var s = _formattedBytes[data[i]];
```

Comment:
you don't really need a lookup table here, e.g.
Each char would become `(char)('0' + data [i])`

Reply:
The result of `_formattedBytes[..]` is a 2-char string. I'm not sure how `(char)('0' + data [i])` is equivalent?

Reply:
that would works if you change to a `Span<char>` since it would require a single assignation per `char`

Reply:
@spouliot I don't get it. If `data[i]` for example is 15, we'll want to get `0F`
How would `'0' + 15` get me `0F`? Anything I'm missing?

Reply:
I mean something like

```csharp
	[Benchmark]
	public string UseSpanWithoutTable()
	{
		using (var algorithm = SHA256.Create())
		{
			var data = algorithm.ComputeHash(Encoding.UTF8.GetBytes(Input));

			var min = Math.Min(data.Length, 16);
			
			Span<char> span = stackalloc char[min * 2];
			int i = 0, j = 0;
			while (i < min) {
				var b = data [i++];
				var nib = (b >> 4);
				span[j++] = nib > 9 ? (char)(nib + 'W') : (char)(nib + '0');
				nib = (b & 0xf);
				span[j++] = nib > 9 ? (char)(nib + 'W') : (char)(nib + '0');
			}
			
			return span.ToString();
		}
	}
```

|              Method |     Mean |   Error |  StdDev |
|-------------------- |---------:|--------:|--------:|
|             RentArr | 587.8 ns | 1.49 ns | 1.24 ns |
|             UseSpan | 591.9 ns | 3.44 ns | 3.21 ns |
| UseSpanWithoutTable | 574.9 ns | 1.60 ns | 1.34 ns |

It's not terribly faster [1] than the use of span but it requires less code (no table) and memory.

[1] to be honest the majority of the time is likely spent on `ComputeHash` and not on the string conversion 😄

Reply:
Thanks for clarifying @spouliot 
I switched to this approach in the last commit